### PR TITLE
Add dependencies doc

### PR DESCRIPTION
**Pull request by linkabi9**
**Prior to creating a pull request, please follow all the steps in the [contributing guide](CONTRIBUTING.md).**

# Description

Created a new doc to track codebase dependencies in plain english that should enable developers to better prepare for the content of the codebase

## Related Issue

- There are no related Github Issues

## Affected services

- [ ] Home
- [ ] News
- [ ] Shop
- [ ] Travel
- [ ] DSP
- [ ] SSP
- [x] ALL

Other:

### CODE REVIEW

**Review by chasinandrew — DISMISSED** [review]

**Comment by chasinandrew** [comment]
Not sure why the pre-commit ci failed, need to look into that. It shouldn't fail on updates to docs

**Review by siddharth-sahoo — DISMISSED** [review]

**Review by siddharth-sahoo — APPROVED** [review]
